src/Files.App/Services/NetworkDrivesService.cs

```diff
@@ -1,73 +1,208 @@
-﻿// Copyright (c) 2024 Files Community
+// Copyright (c) 2024 Files Community
 // Licensed under the MIT License. See the LICENSE.
 
-using Files.App.Utils.Shell;
-using Files.Core.Storage.LocatableStorage;
+using System.Runtime.InteropServices;
+using System.Text;
+using Vanara.InteropServices;
 using Vanara.PInvoke;
 using Vanara.Windows.Shell;
+using static Vanara.PInvoke.AdvApi32;
+using static Vanara.PInvoke.Mpr;
 
 namespace Files.App.Services
 {
-	public sealed class NetworkDrivesService : INetworkDrivesService
+	public sealed class NetworkDrivesService : ObservableObject, INetworkDrivesService
```

Comment:
Is it ok for a service class to be an ObservableObject? I think service classes should be completely separated from views and observable properties should be held by view models.

Reply:
We have some properties like this such as Removable drives, Quick access folders,  recent files/folders, wsl drives, network drives.

What do you think an idea to create a comprehensive view model that holds such drives and folders?

Reply:
Ideally, a view model should exist tied to a view, and one should avoid creating a view model that is inconsistent with the view's lifecycle.

Reply:
Those drives and folders are used only for sidebars and widgets. Sidebar has the longest lifecycle, which ends when user kills the process, so making that view model singleton pattern is totally fine. while recent files are used only for widgets, the view model of the recent widget user control is singleton pattern.

Reply:
I still don't think the ViewModel should be registered as a singleton because the lifecycle of SideBar may change in the future.
Anyway, I don't think it is a good idea to make NetworkDrivesService an ObservableObject.

Reply:
Ok, I'm fine with that.